CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -1497,7 +1494,9 @@ protected function postProcessMembership(
           date('YmdHis'), $membershipParams['cms_contactID'] ?? NULL,
           $customFieldsFormatted,
           $numTerms, $membershipID, $pending,
-          $contributionRecurID, $membershipSource, $isPayLater, $memParams, $membershipContribution,
+          $contributionRecurID, $membershipSource, $isPayLater,
+          $this->_params['campaign_id'] ?? ($this->_values['campaign_id'] ?? NULL),
```

Comment:
Could probably even ditch it & use `$this->getContributionPageValue('campaign_id')` in the next function down - or do we think it might be possible to have campaign_id in a profile (probably they would do something weird & prefix it anyway :-)) - if so then the next level down could still do

`$this->getSubmittedValue('campaign_id') ?: $this->getContributionPageValue('campaign_id')`

I'm pretty sure I fixed these pages so that `getSubmittedValue()` also works on the `Confirm` page